xray/commands/audit/sca/java/mvn.go

```diff
@@ -1,98 +1,146 @@
 package java
 
 import (
+	_ "embed"
 	"errors"
 	"fmt"
-	"github.com/jfrog/jfrog-cli-core/v2/artifactory/utils"
-	"github.com/jfrog/jfrog-cli-core/v2/utils/config"
-	"github.com/jfrog/jfrog-cli-core/v2/utils/coreutils"
-	mvnutils "github.com/jfrog/jfrog-cli-core/v2/utils/mvn"
-	"github.com/jfrog/jfrog-client-go/auth"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
 	"github.com/jfrog/jfrog-client-go/utils/io/fileutils"
 	"github.com/jfrog/jfrog-client-go/utils/log"
 	xrayUtils "github.com/jfrog/jfrog-client-go/xray/services/utils"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
 )
 
-func buildMvnDependencyTree(params *DependencyTreeParams) (modules []*xrayUtils.GraphNode, uniqueDeps []string, err error) {
-	buildConfiguration, cleanBuild := createBuildConfiguration("audit-mvn")
-	defer func() {
-		err = errors.Join(err, cleanBuild())
-	}()
+const (
+	mavenDepTreeJarFile    = "maven-dep-tree.jar"
+	mavenDepTreeOutputFile = "mavendeptree.out"
+	mavenDepTreeVersion    = "1.0.2"
+	TreeCmd                = "tree"
+	ProjectsCmd            = "projects"
+	settingsXmlFile        = "settings.xml"
+)
+
+var settingsXmlTemplate = `<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
There is a chance that the username is not provided or can't be extracted from the reference token. In that case, I'd recommend using the following server configuration:

```xml
<server>
  <id>artifactory</id>
  <configuration>
    <httpHeaders>
      <property>
        <name>Authorization</name>
        <value>Bearer %s</value>
      </property>
    </httpHeaders>
  </configuration>
</server>
```

You can also use the [basic authentication](https://www.wikiwand.com/en/Basic_access_authentication#Client_side) in this type of settings.xml:

```xml
<server>
  <id>artifactory</id>
  <configuration>
    <httpHeaders>
      <property>
        <name>Authorization</name>
        <value>Basic base64(username:password)</value>
      </property>
    </httpHeaders>
  </configuration>
</server>
```